skimage/color/colorconv.py

```diff
@@ -790,6 +790,12 @@ def rgb2gray(rgb):
     """
 
     if rgb.ndim == 2:
+        warn('The behavior of rgb2gray will change in scikit-image 0.19. '
```

Comment:
I think we need a multichannel argument to allow the user to use the new behavior today

Reply:
@hmaarrfk I thought about that, but then that suggests either that we will remove the multichannel argument in the future (another breaking change :man_facepalming:) or that it will do nothing in the future (not great either), or that we will continue to support multichannel forever (which as far as I know is false at this point). :thinking:

Reply:
This was always the blocker for this deprecation. 

I don't think this new behavior is too critical, so we can avoid the introduction of the new parameter.

Reply:
Yes, the warning message is explicit enough to not make the user confused: old users are not disturbed since the old behavior is kept, and users needing the new behavior need to wait version 0.19.
To avoid this warning, they only have to check to not provide 2D arrays as input...
Same argument is valid for #4440

Reply:
@rfezzani I got confused between the two PRs. In this one, it's fine, as the 2D use case isn't that compelling. In the other one, people who want to convert 3D images to RGB where the last dimension has shape 3 have no way to use the library in a future-proof way.

Reply:
>  In the other one, people who want to convert 3D images to RGB where the last dimension has shape 3 have no way to use the library in a future-proof way.

We may advise to add a dummy dimension and to remove it after conversion
`rgb2gray(rgb[np.newaxis, ...])[0, ...]`

Reply:
Again confused :smile:! `rgb2gray(rgb[np.newaxis, ...])[0, ...]` is valid in this PR, and `gray2rgb(image[np.newaxis, ...])[0, ...]` also if we remove this [line](https://github.com/scikit-image/scikit-image/pull/4440/files#diff-9f4dedd98e7cca606dec9f0b792c1d94R835).

Reply:
Ah! Advising reshapes before and after is a great idea! :tada: Could you add the correct one in each case? =)

`gray2rgb(image.ravel()).reshape(image.shape + (3,))` will work for nD gray2rgb.

Reply:
What about adding these advises when the 3D support is effectively allowed (in #4418)? :wink: